editorial: CI for roleInfo.js

pkra opened the pull request:
Adds github actions and scripts to enable automatic checking and updating of roleInfo.js

- common/script/buildRoleInfo.js
  - currently very hacky way of running aria.js using linkedom (see #2501 for plans on removing hackery)
- roleInfo.yml: updates roleInfo.js on pushes to main
  - standard nodejs setup
  - installs linkedom and runs buildRoleInfo.js to update roleInfo
- roleInfoCheck.yml: checks PRs if they introduce changes to roleInfo.js
   - similar to roleInfo.yml but only reports on git diff (no commit)

Part of  #2501


<!--
    This comment and the below content is programmatically generated.
    You may add a comma-separated list of anchors you'd like a
    direct link to below (e.g. #idl-serializers, #idl-sequence):

    Don't remove this comment or modify anything below this line.
    If you don't want a preview generated for this pull request,
    just replace the whole of this comment's content by "no preview"
    and remove what's below.
-->
***
<a href="https://pr-preview.s3.amazonaws.com/w3c/aria/pull/2625.html" title="Last updated on Sep 25, 2025, 10:19 AM UTC (fe811fa)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/aria/2625/42350b4...fe811fa.html" title="Last updated on Sep 25, 2025, 10:19 AM UTC (fe811fa)">Diff</a>

pkra committed:
editorial: CI to update roleInfo

First working draft of a CI approach.
- common/script/buildRoleInfo.js
  - currently very hacky way of running aria.js using linkedom
- roleInfo.yml
  - standard nodejs setup
  - installs linkedom and runs buildRoleInfo.js to update roleInfo

Part of  #2501

spectranaut reviewed (approved):
Ok, well, as you say, this is pretty silly, but I'm fine with it as an intermediate step!

I looked at the code but not enough to try to think of suggesting anything differently. Let me know if you want me to think harder about it.

Edit: oh now I see maybe you want to answer these other questions before landing. I will come back to this and think about it.

pkra committed:
editorial: run roleInfo.yml on push to main

As per editors' decision.

pkra committed:
editorial: add roleInfoCheck.yml (for PRs)

As per editor's decision, this action fails when a PR creates changes in roleInfo.js;
it is meant to trigger review.

pkra committed:
editorial: roleInfo.yml shouldn't fail when there are no changes

Prevents "nothing to commit" errors when there are no changes.

pkra committed:
editorial: separate names for roleInfo actions

pkra committed:
editorial: roleInfoCheck.yml run prettier

Runs prettier to match current output.

pkra committed:
editorial: roleInfoCheck.yml fix prettier

Uses npx to find prettier.

pkra committed:
editorial: roleInfoCheck.yml run prettier on the correct file

Sigh.

pkra committed:
editorial: buildRoleInfo.js remove WIP comments
